### ISSUE TITLE
Accessibility: A11y_Live Chat Wedget_Upload File_ Keyboard: Keyboard focus is not in logical order after upload any file.

### ISSUE THREAD

**Issue by compulim** [microsoft/BotFramework-WebChat#3056]
Copied from #1750405.

[1750405_A11y_Live Chat Wedget_Upload File_ Keyboard.zip](https://github.com/microsoft/BotFramework-WebChat/files/4397038/1750405_A11y_Live.Chat.Wedget_Upload.File_.Keyboard.zip)

“Check out Accessibility Insights! - Identify accessibility bugs before check-in and make bug fixing faster and easier.”

Environment Details:

Application Name: Live Chat Wedget

#URL: https://xxx.blob.core.windows.net/portals/democ2occatest4.html

Browser Versions: Microsoft Edge Version 80.0.361.69 (Official build) (64-bit)

Windows Version: Windows10

Chrome Version: Version 80.0.3987.149 (Official Build) (64-bit)

 

Repro Steps:

1)Launch the application url : https://xxx.blob.core.windows.net/portals/democ2occatest4.html

2)welcome to live chat support Screen get open.

4)Navigate to "Chat Now button" and click on it.

5)"Chat Form" screen get open.

6)Fill all the mandatory detail and then Click on "Submit" Button

7)now "Chat Window" get open.

8)Navigate to "Upload File Button" and click on it then select any file(Once Smaller than 5MB,Then Greater than 5MB)

Actual:

Keyboard focus is not in logical order after upload any file.

Same issue observed with "Let's Chat Upload File" section

Same issue is also observed with "Chrome".

Expected:

Keyboard focus should be in logical order after after selected any file. Focus should be goes to alert type dismiss button.

User Impact:

Keyboard users will get confused while navigating screen due to focus order of  Keyboard.

 

Recommendations:

Please refer to below URL which is repository for accessibility bug fixes. ​

https://microsoft.sharepoint.com/teams/msenable/mas/Pages/browse-fixes.aspx​

 

MAS Reference:

MAS2.4.3:

https://microsoft.sharepoint.com/teams/msenable/_layouts/15/WopiFrame.aspx?sourcedoc={0de7fbe1-ad7e-48e5-bcbb-8d986691e2b9}

 

MAS1.3.2:

https://microsoft.sharepoint.com/:w:/r/teams/msenable/_layouts/15/WopiFrame.aspx?sourcedoc={1ee227a7-e3c5-47b8-ab18-48ca8595077b}

**Comment by compulim** [comment]
## Repro steps:

1. In LCW, upload a file larger than 5 MB, then click Upload
2. LCW will pop up a notification saying this file is too big

### Expected

User focus should move to the dismiss button on the notification.

### Actual

User focus is not moved.

## Additional context

Since the notification toast is *not modal*, we are not moving the focus over there.

We need to look at WCAG/MAS recommendation on changing focus on non-modal elements.

**Comment by compulim** [comment]
According to WAI-ARIA, we should not move the focus, https://www.w3.org/TR/wai-aria-practices/#alert.

The notification toast is providing a time-sensitive but not workflow interrupting experience. We are not implementing alert dialog experience (a.k.a. modal), thus, we should not change focus.

> An alert is an element that displays a brief, important message in a way that attracts the user's attention without interrupting the user's task. Dynamically rendered alerts are automatically announced by most screen readers, and in some operating systems, they may trigger an alert sound. It is important to note that, at this time, screen readers do not inform users of alerts that are present on the page before page load completes.
> 
> Because alerts are intended to provide important and potentially time-sensitive information without interfering with the user's ability to continue working, it is crucial they do not affect keyboard focus. The alert dialog is designed for situations where interrupting work flow is necessary.
>
> It is also important to avoid designing alerts that disappear automatically. An alert that disappears too quickly can lead to failure to meet WCAG 2.0 success criterion 2.2.3. Another critical design consideration is the frequency of interruption caused by alerts. Frequent interruptions inhibit usability for people with visual and cognitive disabilities, which makes meeting the requirements of WCAG 2.0 success criterion 2.2.4 more difficult.
